PGM/src/main/java/tc/oc/pgm/commands/MapCommands.java

```diff
@@ -45,20 +46,47 @@
     @Command(
         aliases = {"maplist", "maps", "ml"},
         desc = "Shows the maps that are currently loaded",
-        usage = "[page]",
+        usage = "[author] [gamemode] [page]",
         min = 0,
-        max = 1,
+        max = 2,
+        flags = "ag",
         help = "Shows all the maps that are currently loaded including ones that are not in the rotation."
     )
     @CommandPermissions("pgm.maplist")
-    public static void maplist(CommandContext args, final CommandSender sender) throws CommandException {
+    public List<String> maplist(CommandContext args, final CommandSender sender) throws CommandException {
         final Set<PGMMap> maps = ImmutableSortedSet.copyOf(new PGMMap.DisplayOrder(), PGM.getMatchManager().getMaps());
-
+        Set<PGMMap> toDisplay = new HashSet<>();
+
+        for (PGMMap map : maps) {
+            if (!args.getFlags().isEmpty() && args.argsLength() == 1) {
+                if (args.hasFlag('a')) {
+                    List<Contributor> authors = map.getInfo().getNamedAuthors();
+                    for (Contributor author : authors) {
+                        if (author.getName().toLowerCase().equalsIgnoreCase(args.getString(0))) {
+                            toDisplay.add(map);
+                        }
+                    }
+                }
+                if (args.hasFlag('g')) {
```

Comment:
right now how this is setup is that either author or gamemode matches, i'm pretty sure you'd want author AND gamemode to match